File: Javascript/Arrays/Eliminacion_elegida.js
/*
Eliminacion en Javascript, Eliminacion de un elemento elegible de un array.
@Autor: Daniel Alejandro Alvarado Velázquez
@grupo: #2-02
@Fecha: 03/09/2026
*/

//Declramos el array
const input_arr = [11, 21, 31, 41, 51, 61];
const elem = 41;
const index = 4;
const elem_borrar = 1;

console.log("Antes de la eliminacion, el array es: ");
for (let i = 0; i < input_arr.length; i++){
    console.log(input_arr[i]);
}

input_arr.splice(index, elem_borrar);       //.splice(indice del objetivo, cuantos elementos borrar desde ese indice)
/*
for (let i = 0; i < input_arr.length; i++){
    if (input_arr[i] == elem){
        input_arr.splice(i, 1);
    }
}
*/
console.log("Despues de la eliminacion, el array es: ");
for (let i = 0; i < input_arr.length; i++){
    console.log(input_arr[i]);
}

File: C#/Arrays/Eliminacion_elegida.cs
/*
Eliminacion en C#, Eliminacion de un elemento elegido de un array.
@Autor: Daniel Alejandro Alvarado Velázquez
@grupo: #2-02
@Fecha: 06/09/2026
*/

using System;
using System.Linq;

namespace Eliminacion_elegida
{
    class Program
    {
        static void Main(String[] args)
        {
            int [] input_arr = {1, 11, 21, 31, 41, 51};
            int elem = input_arr[3];

            Console.WriteLine("Antes de la eliminacion, el array es: ");
            foreach (int elemento in input_arr){
            Console.WriteLine(elemento);
            }

            input_arr = input_arr.Where(n => n != elem).ToArray();        //arreglo = arreglo.where(n(n= elemento) => elemento != elemento a borrar).toArray()
            //se modifica el arreglo 

            Console.WriteLine("Despues de la eliminacion, el array es: ");
            foreach (int elemento in input_arr){
            Console.WriteLine(elemento);
            }
        }
    }
}